spring-integration-ws/src/main/java/org/springframework/integration/ws/MarshallingWebServiceOutboundGateway.java

```diff
@@ -107,39 +116,39 @@ private void configureMarshallers(Marshaller marshaller, Unmarshaller unmarshall
 		if (unmarshaller == null) {
 			Assert.isInstanceOf(Unmarshaller.class, marshaller,
 					"Marshaller [" + marshaller + "] does not implement the Unmarshaller interface. " +
-					"Please set an Unmarshaller explicitly by using one of the constructors that accepts " +
-					"both Marshaller and Unmarshaller arguments.");
+							"Please set an Unmarshaller explicitly by using one of the constructors that accepts " +
+							"both Marshaller and Unmarshaller arguments.");
 			unmarshaller = (Unmarshaller) marshaller;
 		}
 
 		Assert.notNull(unmarshaller, "unmarshaller must not be null");
-		this.marshaller = marshaller;
-		this.unmarshaller = unmarshaller;
-	}
 
-	private final class MarshallingRequestMessageCallback extends RequestMessageCallback {
+		getWebServiceTemplate().setMarshaller(marshaller);
+		getWebServiceTemplate().setUnmarshaller(unmarshaller);
+	}
 
-		MarshallingRequestMessageCallback(WebServiceMessageCallback requestCallback,
-				Message<?> requestMessage) {
-			super(requestCallback, requestMessage);
-		}
+	@Override
+	public String getComponentType() {
+		return "ws:outbound-gateway(marshaling)";
```

Comment:
He-he! Existing code for a while already 😄 

But thanks for the care on merge!